.ci/new_ci.jenkinsfile

```diff
@@ -0,0 +1,435 @@
+}
+
+def getModernImages(String pr_number, String registry) {
+    Map<String, String> envValues = [:]
+    stage('Modern - Parse environment file') {
+        envValues = readEnvValues("modern/.env")
+    }
+
+    List<Map> images = []
+    stage('Modern - Generate build matrix') {
+        List<String> gccVersions = envValues["GCC_VERSIONS"].split(",").collect({String version -> return getVersionMajor(version)})
+        List<String> clangVersions = envValues["CLANG_VERSIONS"].split(",").collect({String version -> return getVersionMajor(version)})
+        List<String> conanVersions = [envValues["CONAN_VERSION"]]
+        List<String> jenkinsAgentVersions = envValues["JENKINS_AGENT_VERSION"].split(",")
+        String cmakeVersion = envValues["CMAKE_VERSION_FULL"]
+        String pythonVersion = envValues["PYTHON_VERSION"]
+        String dockerUsername = envValues["DOCKER_USERNAME"]
+        String dockerhubUsername = envValues["DOCKERHUB_USERNAME"]
+        String distro = envValues["DISTRO"]
+        String distroVersion = envValues["DISTRO_VERSION"]
+        String libstdcppVersion = envValues["LIBSTDCPP_VERSION"]
+        String libstdcppPatchVersion = envValues["LIBSTDCPP_PATCH_VERSION"]
+        String libstdcppMajorVersion = envValues["LIBSTDCPP_MAJOR_VERSION"]
+
+        List<String> allImages = []
+
+        conanVersions.each({ String conanVersion ->
+            allImages.add("${dockerUsername}/base-${distro}:${conanVersion}")
+        })
+
+        List<String> compilerImages = []
+        gccVersions.each({ String gccVersion ->
+            String image = "${dockerUsername}/gcc${gccVersion}-${distro}"
+            compilerImages.add(image)
+        })
+
+        clangVersions.each({ String clangVersion ->
+            String image = "${dockerUsername}/clang${clangVersion}-${distro}"
+            compilerImages.add(image)
+        })
+
+        List<String> conanImages = []
+        conanVersions.each({ String conanVersion ->
+            compilerImages.each({ String compilerImage ->
+                conanImages.add("${compilerImage}:${conanVersion}")
+            })
+        })
+
+        allImages.addAll(conanImages.collect())
+        conanImages.each({String conanImage ->
+            def (name, conanVersion) = conanImage.split(':')
+            allImages.add("${name}-builder:${conanVersion}")
+
+            jenkinsAgentVersions.each({String jenkinsAgentVersion ->
+                allImages.add("${name}-jenkins:${conanVersion}-${jenkinsAgentVersion}")
+            })
+        })
+        echo "All images: ${allImages}"
+
+        allImages.each({ String image ->
+            Map imageData = ["buildArgs": [:], "target": null]
+            imageData["name"] = "${image}-#${pr_number}"
+            imageData["compiler"] = image.contains("gcc") ? "gcc" : "clang"  // remove?
+            String compilerName = imageData["compiler"].toUpperCase()
+            String compilerVersionSubStr = image.split(imageData["compiler"]).last()
```

Comment:
Same here, map fits better and complicated split last from a variable.

Reply:
totally right, this is the most important part to improve of the changes in this PR. I would like to focus on a new PR

Reply:
Done in recent commit